File: backend/CustomerUpdate.API/Controllers/HealthController.cs
using Microsoft.AspNetCore.Mvc;
using CustomerUpdate.API.DTOs;

namespace CustomerUpdate.API.Controllers;

[ApiController]
[Route("api/[controller]")]
public class HealthController : ControllerBase
{
    [HttpGet]
    public ActionResult<ApiResponse<object>> GetHealth()
    {
        return Ok(new ApiResponse<object>
        {
            Success = true,
            Message = "API is healthy",
            Data = new 
            { 
                status = "Healthy",
                timestamp = DateTime.UtcNow,
                version = "1.0.0",
                environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Development"
            }
        });
    }
} 

File: backend-java/src/main/java/com/example/customerupdate/controller/HealthController.java
package com.example.customerupdate.controller;

import com.example.customerupdate.dto.ApiResponse;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/health")
public class HealthController {
    @Value("${spring.profiles.active:development}")
    private String environment;

    @GetMapping
    public ApiResponse<Object> getHealth() {
        Map<String, Object> data = new HashMap<>();
        data.put("status", "Healthy");
        data.put("timestamp", Instant.now());
        data.put("version", "1.0.0");
        data.put("environment", environment);
        ApiResponse<Object> response = new ApiResponse<>();
        response.setSuccess(true);
        response.setMessage("API is healthy");
        response.setData(data);
        return response;
    }
} 